bfh/fields.py

```diff
@@ -63,7 +64,10 @@ def __get__(self, instance, cls=None):
         return instance.__dict__.get(self.field_name)
 
     def __set__(self, instance, value):
-        instance.__dict__[self.field_name] = value
+        if value is None:
```

Comment:
And i *think* the implementation that matches all those behaviors looks something like:

"in `__get__`, if value is None and a default exists, put the default in the value (calling here if it's a callable)"